### PR TITLE
Updates downstreamed from Stratum project

### PR DESCRIPTION
**Pull request by ffoulkes**
Cumulative updates downstreamed from the Stratum project.

To build with cmake, you will need the `dgf/downstream` branch of `networking-recipe`.

### CODE REVIEW

File: stratum/lib/security/credentials_manager.cc [hunk 1]
```diff
@@ -42,14 +37,57 @@ using ::grpc::experimental::TlsServerCredentialsOptions;
 
 constexpr unsigned int CredentialsManager::kFileRefreshIntervalSeconds;
 
-CredentialsManager::CredentialsManager() {}
+namespace {
+
+const std::map<std::string, grpc_ssl_client_certificate_request_type>
+    client_cert_req_type_map = {
+        {"NO_REQUEST", GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE},
```
Comment (5abeel) [inline stratum/lib/security/credentials_manager.cc:44]: Documentation will need to be updated to reflect these name changes

File: stratum/lib/security/credentials_manager.cc [hunk 2]
```diff
@@ -63,95 +101,73 @@ CredentialsManager::GenerateExternalFacingClientCredentials() const {
...
+      LOG(WARNING) << "No key/certificate files specified; "
+                      "cannot initialize server credentials.";
     } else {
-      LOG(WARNING) << "No certificate/key files provided, using insecure "
-                      "server credentials!";
+      LOG(WARNING) << "No key/certificate files specified; "
+                      "using insecure server credentials!";
       server_credentials_ = ::grpc::InsecureServerCredentials();
     }
+  } else if (secure_only_ && !ServerFilesOk()) {
```
Comment (5abeel) [inline stratum/lib/security/credentials_manager.cc:121]: Code looks ok, but should be tested to make sure that there is no fallbacks to insecure modes

**Review by 5abeel — APPROVED** [review]
LGTM

**Review by nupuruttarwar — APPROVED** [review]
LGTM

**Review by aashishkuma — APPROVED** [review]
LGTM

**Pull request merged**